src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Check.java

```diff
@@ -3512,7 +3512,7 @@ boolean isOverrider(Symbol s) {
     protected boolean isTypeAnnotation(JCAnnotation a, boolean isTypeParameter) {
         List<Attribute> targets = typeAnnotations.annotationTargets(a.annotationType.type.tsym);
         return (targets == null) ?
-                false :
+                (source.compareTo(Source.JDK14) >= 0 && isTypeParameter) :
```

Comment:
Usually this kind of test is done using a constant in the Feature enum; I'll let Vicente or other engineer who focuses on javac weigh in on whether or not the direct comparison is acceptable in this case.

Reply:
right I agree we should add a new member to the Feature enum

Reply:
Will do, do you have an preferences for the name of the new constant? Maybe `NO_TARGET_ANNOTATIONS_ARE_APPLICABLE_TO_TYPE_PARAMETER_DECLARATIONS`?

Reply:
I went with `NO_TARGET_ANNOTATION_APPLICABILITY`, I'm happy to rename it